src/Compilers/CSharp/Portable/Syntax/LookupPosition.cs

```diff
@@ -320,7 +320,7 @@ private static SyntaxToken GetFirstIncludedToken(StatementSyntax statement, bool
                 case SyntaxKind.ReturnStatement:
                     return ((ReturnStatementSyntax)statement).ReturnKeyword;
                 case SyntaxKind.SwitchStatement:
-                    return ((SwitchStatementSyntax)statement).OpenBraceToken;
+                    return ((SwitchStatementSyntax)statement).Expression.GetFirstToken();
```

Comment:
Note, this change doesn't affect the binding, it only affects the process of looking up an enclosing binder within SemanticModel.
Before this change, only things inside ```{}``` were considered by SemanticModel to be in scope owned by a switch statement. After this change, the switch expression is also considered to be in scope owned by the switch. There are two scopes owned by a switch, the first spans expression and all switch sections, the second spans just switch sections and is nested into the first. Binder.GetBinder API returns binder representing the second scope, it never returns binder for the first scope. The change in MemberSemanticModel.AdjustBinderForPositionWithinStatement takes care of that. 
There was a bug in the binding as well, where a binder for the second scope was used to bind a switch expression, making locals declared in switch sections visible there. That was addressed by the change in Binder.BindPatternSwitch.